player/command.c

```diff
@@ -6521,7 +6521,7 @@ static void cmd_apply_profile(void *p)
     int mode = cmd->args[1].v.i;
     if (mode == 0) {
         cmd->success = m_config_set_profile(mpctx->mconfig, profile, 0) >= 0;
-    } else {
+    } else if (!mp_is_shutting_down(mpctx)) {
```

Comment:
Does this work with watch later option save? I think it should safe while closing file, so I don't expect profile change there, but just thinking.

Reply:
This only makes `profile-restore=copy(-equal)` be a no-op if the evaluation of profile-cond changes while we're shutting down. The only way this could affect watch-later is if you have `save-position-on-quit=no` inside your auto profile.